test/test_torch.py

```diff
@@ -5969,129 +5908,60 @@ def test_geqrf(self):
+            else:
+                self.assertLessEqual(b.dist(A.mm(x)), 4e-12)
 
     @skipIfNoLapack
     def test_triangular_solve(self):
         self._test_triangular_solve(self, lambda t: t)
 
     @staticmethod
     def _test_triangular_solve_batched(self, cast):
-        def triangular_solve_test_helper(A_dims, b_dims, cast, upper, unitriangular):
-            A = cast(torch.randn(*A_dims))
-            A = A.triu() if upper else A.tril()
-            if unitriangular:
-                A.diagonal(dim1=-2, dim2=-1).fill_(1.)
-            b = cast(torch.randn(*b_dims))
-            return A, b
+        from common_utils import triangular_solve_test_helper
 
-        for upper, transpose, unitriangular in product([True, False], repeat=3):
-            # test against triangular_solve: one batch with all possible arguments
-            A, b = triangular_solve_test_helper((1, 5, 5), (1, 5, 10), cast, upper, unitriangular)
-            x_exp = torch.triangular_solve(b.squeeze(0), A.squeeze(0),
-                                           upper=upper, transpose=transpose, unitriangular=unitriangular)[0]
-            x = torch.triangular_solve(b, A,
-                                       upper=upper, transpose=transpose, unitriangular=unitriangular)[0]
-            self.assertEqual(x, x_exp.unsqueeze(0))
-
-            # test against triangular_solve in a loop: four batches with all possible arguments
-            A, b = triangular_solve_test_helper((4, 5, 5), (4, 5, 10), cast, upper, unitriangular)
+        def triangular_solve_batch_helper(A_dims, b_dims, cast, upper, unitriangular, transpose):
+            b, A = triangular_solve_test_helper(A_dims, b_dims, cast, upper, unitriangular)
             x_exp_list = []
-            for i in range(4):
-                x_exp = torch.triangular_solve(b[i], A[i],
-                                               upper=upper, transpose=transpose, unitriangular=unitriangular)[0]
-                x_exp_list.append(x_exp)
-            x_exp = torch.stack(x_exp_list)
-
-            x = torch.triangular_solve(b, A, upper=upper, transpose=transpose, unitriangular=unitriangular)[0]
-            self.assertEqual(x, x_exp)
-
-            # basic correctness test
-            A, b = triangular_solve_test_helper((3, 5, 5), (3, 5, 10), cast, upper, unitriangular)
-            x = torch.triangular_solve(b, A, upper=upper, transpose=transpose, unitriangular=unitriangular)[0]
+            for i in range(b_dims[0]):
+                x_exp_list.append(torch.triangular_solve(b[i], A[i], upper=upper,
+                                                         unitriangular=unitriangular, transpose=transpose)[0])
+            x_exp = torch.stack(x_exp_list)  # Stacked output
+            x_act = torch.triangular_solve(b, A, upper=upper,
+                                           unitriangular=unitriangular, transpose=transpose)[0]  # Actual output
+            self.assertEqual(x_exp, x_act)  # Equality check
```

Comment:
nit: switch the order of the args. Usually equality checks are `self.assertEqual(obtained_result, expected)`

Reply:
Sure, thank you.